### PR TITLE
Fix citizen request fulfilling sometimes messing up nbt

### LINKED ISSUE #5264: Knight changed enchantment of given sword (fulfilling requeset)
### Minecolonies version

Minecolonies: unspecified, commit 129518831bba86439c1c7c4a91057f1d1db68299
Structurize: 0.10.324-ALPHA
Minecraft: 1.15.2
Forge: 31.2.20
Single-player/survival gamemode

### Description
1. Knight is requesting a sword.
2. Have two iron swords in inventory enchanted with:
    - Sharpness II, Unbreaking III
    - Unbreaking I, Bane of Arthropods I
 3. Go to the Knight and fulfill the request (press `Fulfill` in request detail view)

### Expected behavior
The Knight gets one of the swords, **without** changing ~~the other swords~~ any enchantment.

### Actual behavior
The Knight got the first sword (Shrapness II, Unbreaking III) but changed its enchantment to Unbreaking I, Bane of Arthropods I - that is, both swords ended up with Unbreaking I, Bane of Arthropods I.

More details in [comment](https://github.com/ldtteam/minecolonies/issues/5264#issuecomment-648918423) below.

### PR DESCRIPTION
**Pull request by ToMe25**
Closes #5264
Closes #
Closes #

# Changes proposed in this pull request:
- Actually check the item nbt of the item to move to the citizen to prevent it from getting changed during the transfer.
- Also remove a completely commented out method, because i thought its in that class, and that made it easier to look around.
- Bump Forge requirement to 31.2.8

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/network/messages/server/colony/citizen/TransferItemsToCitizenRequestMessage.java [hunk 1]
```diff
@@ -126,7 +126,7 @@ protected void onExecute(final NetworkEvent.Context ctxIn, final boolean isLogic
             int amountToRemoveFromPlayer = amountToTake - ItemStackUtils.getSize(remainingItemStack);
             while (amountToRemoveFromPlayer > 0)
             {
-                final int slot = InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(player.inventory), stack -> stack.isItemEqual(itemStack));
+                final int slot = InventoryUtils.findFirstSlotInItemHandlerWith(new InvWrapper(player.inventory), stack -> stack.isItemEqual(itemStack) && ItemStack.areItemStackTagsEqual(stack, itemStack));
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/network/messages/server/colony/citizen/TransferItemsToCitizenRequestMessage.java]: Just use ItemStackUtils equal method .

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/network/messages/server/colony/citizen/TransferItemsToCitizenRequestMessage.java]: Its like this in containers, so i just stole it from there, is that wrong?

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/network/messages/server/colony/citizen/TransferItemsToCitizenRequestMessage.java]: also there is no ItemStackUtils.equal(), or am i not seeing some third ItemStackUtils?

Comment (ToMe25) [reply src/main/java/com/minecolonies/coremod/network/messages/server/colony/citizen/TransferItemsToCitizenRequestMessage.java]: done.

**Pull request merged**